src/test/java/org/folio/rest/api/HridSettingsIncreaseMaxValueMigrationTest.java

```diff
@@ -0,0 +1,139 @@
+package org.folio.rest.api;
+
+import static org.folio.rest.api.StorageTestSuite.TENANT_ID;
+import static org.folio.rest.api.StorageTestSuite.getVertx;
+
+import java.util.List;
+
+import org.folio.rest.persist.PostgresClient;
+import org.folio.util.ResourceUtil;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonArray;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+@RunWith(VertxUnitRunner.class)
+public class HridSettingsIncreaseMaxValueMigrationTest extends TestBase {
+  private static final String CREATE_SEQUENCE_SQL = loadCreateSequenceSqlFile();
+  private static final String ALTER_SEQUENCE_SQL = loadAlterSequenceSqlFile();
+  private static final String INSTANCES_SEQ = "hrid_instances_seq";
+  private static final String HOLDINGS_SEQ = "hrid_holdings_seq";
+  private static final String ITEMS_SEQ = "hrid_items_seq";
+
+  @Test
+  public void canSaveCurrentValueOfSequences(TestContext testContext) {
+    reCreateSequences()
+      // Increment sequences and and set a new value
+      .compose(notUsed -> incrementSequence(INSTANCES_SEQ, 10))
+      .map(hrid -> testContext.assertEquals(10L, hrid))
+      .compose(notUsed -> incrementSequence(HOLDINGS_SEQ, 7))
+      .map(hrid -> testContext.assertEquals(7L, hrid))
+      .compose(notUsed -> setValue(ITEMS_SEQ, 999_999_99L))
+      .compose(notUsed -> incrementSequence(ITEMS_SEQ))
+      .map(hrid -> testContext.assertEquals(999_999_99L, hrid))
+      // Alter sequences
+      .compose(notUsed -> alterSequences())
+      // Assert that current value has not been changed
+      .compose(notUsed -> incrementSequence(INSTANCES_SEQ))
+      .map(hrid -> testContext.assertEquals(11L, hrid))
+      .compose(notUsed -> incrementSequence(HOLDINGS_SEQ))
+      .map(hrid -> testContext.assertEquals(8L, hrid))
+      .compose(notUsed -> incrementSequence(ITEMS_SEQ))
+      .map(hrid -> testContext.assertEquals(1_000_000_00L, hrid))
+      // Handle test result
+      .setHandler(testContext.asyncAssertSuccess());
+  }
+
+  @Test
+  public void canSetBigIntValue(TestContext testContext) {
```

Comment:
Why is this test needed? Do the regular tests not try to set a sequence value higher than the previous limit (and bigger than an integer)?

I think it would be preferable to have regular tests that successfully get a large (nearly maximum) value for the HRID.

Reply:
Yes, we have such test `canGetNextHridWhenStartNumberIsLong`. This test to check it on DB level.

Reply:
I personally don't think we need a separate test for it here as well.

If we could express it without integration, then I might understand the value of it, however this still requires external integration so I'm not sure it is worth it.

Reply:
Ok, removed it, because we have the integration test.